src/aind_behavior_experiment_launcher/behavior_launcher/slims_picker.py

```diff
@@ -0,0 +1,270 @@
+
+        Returns:
+            SlimsClient: slims client instance.
+
+        Raises:
+            Exception: If invalid credentials or error reading from slims
+        """
+
+        try:
+            logger.info("Attempting to connect to Slims")
+            slims_client = SlimsClient(
+                url=url,
+                username=username if username else os.environ["SLIMS_USERNAME"],
+                password=password if password else os.environ["SLIMS_PASSWORD"],
+            )
+            slims_client.fetch_model(models.SlimsMouseContent, barcode="00000000")
+
+        except exceptions.SlimsAPIException as e:
+            if "Status 401 – Unauthorized" in str(e):  # catch error if username and password are incorrect
+                raise Exception(
+                    f"Exception trying to read from Slims: {e}.\n"
+                    f" Please check credentials:\n"
+                    f"Username: {username if username else os.environ['SLIMS_USERNAME']}\n"
+                    f"Password: {password if password else os.environ['SLIMS_PASSWORD']}"
+                )
+            else:
+                raise Exception(f"Exception trying to read from Slims: {e}.\n")
+
+        logger.info("Successfully connected to Slims")
+
+        return slims_client
+
+    @property
+    def slims_mouse(self) -> models.mouse:
+        """
+        Returns slims mouse model being used to load session
+
+        Returns:
+            models.Mouse: slims mouse model object
+        """
+
+        return self._slims_mouse
+
+    @property
+    def slims_session(self) -> models.behavior_session.SlimsBehaviorSession:
+        """
+        Returns slims session model being used to load task logic
+
+        Returns:
+            models.behavior_session.SlimsBehaviorSession: slims session model object
+        """
+
+        return self._slims_session
+
+    def pick_rig(self) -> _R:
+        """
+        Prompts the user to provide a rig name.
+
+        Returns:
+            _R: The selected rig configuration.
+
+        Raises:
+            SlimsRecordNotFound: If no rig is found in Slims or an invalid choice is made.
+        """
+
+        while True:
+            instruments = self.slims_client.fetch_models(models.SlimsInstrument)[-100:]
+            rig = None
+            while rig is None:
+                rig = self.ui_helper.input("Enter rig name: ")
+                if rig == "":
+                    rig = self.ui_helper.prompt_pick_from_list(
+                        [instrument.name for instrument in instruments],
+                        prompt="Choose a rig:",
+                        allow_0_as_none=True,
+                    )
+                try:
+                    slims_rig = self.slims_client.fetch_model(models.SlimsInstrument, name=rig)
+                    return self.launcher.rig_schema_model(rig_name=slims_rig.name)
```

Comment:
Add a todo to attempt to infer the rig id with env variables. This relies on slims having a stable nomenclature for rigs